nameserver/nameserver.go

```diff
@@ -203,11 +203,30 @@ func (n *Nameserver) receiveGossip(msg []byte) (mesh.GossipData, mesh.GossipData
 		return nil, nil, fmt.Errorf("clock skew of %d detected", delta)
 	}
 
+	gossip.Entries.filter(func(e *Entry) bool {
+		return n.isKnownPeer(e.Origin)
+	})
+
 	n.Lock()
 	defer n.Unlock()
 
+	// Check entries claiming to originate from us against our current data
 	gossip.Entries.filter(func(e *Entry) bool {
-		return n.isKnownPeer(e.Origin)
+		if e.Origin == n.ourName {
+			n.infof("Received entry like mine: %v", e)
+			if ourEntry, ok := n.entries.findEqual(e); ok {
+				if ourEntry.Version < e.Version ||
+					(ourEntry.Version == e.Version && ourEntry.Tombstone != e.Tombstone) {
+					// Take our version of the data, but make the version higher than the incoming
+					higherVersion := e.Version
+					*e = *ourEntry
+					e.Version = higherVersion + 1
```

Comment:
ah, I completely misread what this was doing. maybe, just maybe, doing `higherVersion := e.Version + 1` would be clearer.